### ISSUE TITLE
Customization for lights

### ISSUE THREAD

**Issue by GiraPomba** [SlavicPotato/ied-dev#35]
It has some settings for FOV.
Is it possible to add new settings for Radius, Colors, etc..?

What is this stuff about armors? Managing equipped armor, hiding, showing etc..? I'm interested lol.

**Comment by SlavicPotato** [comment]
It's possible, for now I've only added stuff that can't be changed on light forms, at some point I might make it possible to configure everything from the UI. This would effectively make using forms redundant, lights could simply be attached to anything. For now, you'll have to stick with making new forms though.  

> What is this stuff about armors? Managing equipped armor, hiding, showing etc..? I'm interested lol.

It's for conditionally equipping outfits. There's another part that allows creating outfit forms in-game which ties the whole thing together, but that's completely unsuitable for the public because it writes to the save (not the skse co-save like plugins normally do).

**Comment by GiraPomba** [comment]
> For now, you'll have to stick with making new forms though

Alright.

> It's for conditionally equipping outfits. There's another part that allows creating outfit forms in-game which ties the whole thing together, but that's completely unsuitable for the public because it writes to the save (not the skse co-save like plugins normally do).
 
Sounds really interesting, will you release the builds here?

**Comment by SlavicPotato** [comment]
> Sounds really interesting, will you release the builds here?

Let me explain what I mean by 'completely unsuitable for the public': 

Once the form data is written, special steps need to be taken to remove it from a save after uninstalling (or just disabling) the plugin and if the user makes a mistake they could end up silently corrupting that save. It's dangerous, completely impractical and what's worse, there's no way for me to safeguard against this happening.

It was a quick and easy way to diversify NPC outfits in my game (without ever having to quit the session), so no thought was put into making it usable for anyone else.

That said, it would still work even without the form manager, but it'd be impossible to edit outfits in-game.

**Comment by GiraPomba** [comment]
Ah, that makes sense. With SKSE plugins that can be installed and uninstalled at will without the need to worry about corruption or other issues, it's best to keep that way lol.

**Comment by GiraPomba** [comment]
Speaking of lights, there are two options to not light water and ground. But what about Grass?
Well, in vanilla there's no light for grass, but it could be done with ENB or the recent Grass Lighting mod.

Don't really know if it's possible to integrate IED's function with the other mods, but just putting here as something to consider.

**Comment by SlavicPotato** [comment]
> Don't really know if it's possible to integrate IED's function with the other mods

It's not possible to control on a per-object basis. These mods would have to provide options for that, I don't think that's likely to happen.

**Comment by GiraPomba** [comment]
Yeah, I figured it would use what the game uses for light. The Grass Lighting mod changes something in the engine to affect grass, which in turn affects the lights created with IED too.
I was just wondering if there was a way to disable grass lightning with IED the same way as the water and ground options, might be useful fote someone, but how IED would tell the mods to not light grass, right? Lol

**Comment by GiraPomba** [comment]
Just a question, since this is related to lights. 
Right now, you can't use light forms that don't have models; you get this error in the log: "couldn't get model params".

> at some point I might make it possible to configure everything from the UI. This would effectively make using forms redundant, lights could simply be attached to anything.

I guess this is what you meant?

**Comment by SlavicPotato** [comment]
>I guess this is what you meant?

I meant something similar to how effect shaders are implemented, a separate area for lights, fully configurable params (stuff that's on forms now) with an option to select the attachment node for the light

**Comment by GiraPomba** [comment]
Oh, that would be cool.
Right now I'm just assigning an "invisible" mesh to the light records to be able to use them with IED an illuminate stuff.

**Comment by SlavicPotato** [comment]
> Right now I'm just assigning an "invisible" mesh to the light records to be able to use them with IED and illuminate stuff.

Imo that's not that much of a PITA, where configurable lights would really come in handy is attaching to any random object want. Right now that's impossible.
